src/crewai/crew.py

```diff
@@ -726,7 +724,14 @@ def _execute_tasks(
                     f"No agent available for task: {task.description}. Ensure that either the task has an assigned agent or a manager agent is provided."
                 )
 
-            self._prepare_agent_tools(task)
+            # Determine which tools to use - task tools take precedence over agent tools
+            tools_for_task = task.tools if task.tools else agent_to_use.tools or []
```

Comment:
I'm actually surprised this `or []` works. If it does, could we safely do something like `task.tools or agent_to_use.tools or []`?